From 8922b2aa12c2917e299a51d42ab6cde3bc8687a1 Mon Sep 17 00:00:00 2001
From: Michal Marek <mmarek@suse.cz>
Date: Mon, 10 Dec 2012 13:22:57 +0100
Subject: [PATCH 1/1] MODSIGN: Add option to not sign modules during build
Patch-mainline: Submitted 2013-01-23

To allow the builder to sign only a subset of modules, or to sign the
modules using a key that is not available on the build machine, add
CONFIG_MODULE_SIG_ALL. If this option is unset, no modules will be
signed during build. The default is 'y', to preserve the current
behavior.

Signed-off-by: Michal Marek <mmarek@suse.cz>
---
 Makefile                |    9 +++++----
 init/Kconfig            |   12 ++++++++++++
 scripts/Makefile.fwinst |    9 +++++----
 3 files changed, 22 insertions(+), 8 deletions(-)

--- a/Makefile
+++ b/Makefile
@@ -733,7 +733,7 @@ endif # INSTALL_MOD_STRIP
 export mod_strip_cmd
 
 
-ifeq ($(CONFIG_MODULE_SIG),y)
+ifdef CONFIG_MODULE_SIG_ALL
 MODSECKEY = ./signing_key.priv
 MODPUBKEY = ./signing_key.x509
 export MODPUBKEY
@@ -743,10 +743,11 @@ mod_sign_cmd = true
 endif
 export mod_sign_cmd
 
-ifeq ($(CONFIG_FIRMWARE_SIG),y)
-fw_sign_cmd = perl $(srctree)/scripts/sign-file -f $(MODSECKEY) $(MODPUBKEY)
-else
 fw_sign_cmd = true
+ifdef CONFIG_MODULE_SIG_ALL
+ifdef CONFIG_FIRMWARE_SIG
+fw_sign_cmd = perl $(srctree)/scripts/sign-file -f $(MODSECKEY) $(MODPUBKEY)
+endif
 endif
 export fw_sign_cmd
 
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1473,6 +1473,18 @@ config MODULE_SIG_FORCE
 	  Reject unsigned modules or signed modules for which we don't have a
 	  key.  Without this, such modules will simply taint the kernel.
 
+config MODULE_SIG_ALL
+	bool "Automatically sign all modules"
+	default y
+	depends on MODULE_SIG
+	help
+	  Sign all modules during make modules_install. Without this option,
+	  modules must be signed manually, using the scripts/sign-file tool.
+
+comment "Do not forget to sign required modules with scripts/sign-file"
+	depends on MODULE_SIG_FORCE && !MODULE_SIG_ALL
+
+
 choice
 	prompt "Which hash algorithm should modules be signed with?"
 	depends on MODULE_SIG
--- a/scripts/Makefile.fwinst
+++ b/scripts/Makefile.fwinst
@@ -29,12 +29,13 @@ installed-mod-fw := $(addprefix $(INSTAL
 installed-fw := $(addprefix $(INSTALL_FW_PATH)/,$(fw-shipped-all))
 installed-fw-dirs := $(sort $(dir $(installed-fw))) $(INSTALL_FW_PATH)/.
 
-ifeq ($(CONFIG_FIRMWARE_SIG),y)
-installed-fw-sig := $(patsubst %,%.sig, $(installed-fw))
-installed-mod-fw-sig := $(patsubst %,%.sig, $(installed-mod-fw))
-else
 installed-fw-sig :=
 installed-mod-fw-sig :=
+ifdef CONFIG_MODULE_SIG_ALL
+ifdef CONFIG_FIRMWARE_SIG
+installed-fw-sig := $(patsubst %,%.sig, $(installed-fw))
+installed-mod-fw-sig := $(patsubst %,%.sig, $(installed-mod-fw))
+endif
 endif
 
 quiet_cmd_fwsig = FWSIG $@
